feat(book): add pagination and OpenAPI

vDmytriv01 opened the pull request:
- Added Pageable support to BookController, BookService, and BookServiceImpl
- Added springdoc-openapi with detailed Operation and Schema annotations for all DTOs
- Updated Maven dependencies and versions (Spring Boot, springdoc)
- Updated endpoints with pagination, sorting

vDmytriv01 committed:
feat(book): add pagination and OpenAPI

- Added Pageable support to BookController, BookService, and BookServiceImpl
- Added springdoc-openapi with detailed Operation and Schema annotations for all DTOs
- Updated Maven dependencies and versions (Spring Boot, springdoc)
- Updated endpoints with pagination, sorting

liliia-ponomarenko commented on src/main/java/com/vdmytriv/bookstoreapp/dto/PartialUpdateBookRequestDto.java:
```diff
@@ -1,16 +1,23 @@
 
 public record PartialUpdateBookRequestDto(
+        @Schema(description = "Title of the book", example = "Updated Title")
         Optional<String> title,
```
```suggestion
        Optional<String> title,
``` 
Why Optional

liliia-ponomarenko reviewed (approved):
Good job! 
One question

vDmytriv01 committed:
feat(book): implement partial update with nullable fields and proper validation

- Replaced Optional fields with nullable types in PartialUpdateBookRequestDto
- Used @PositiveOrZero instead of @Min for BigDecimal price validation
- Added @BeanMapping with NullValuePropertyMappingStrategy.IGNORE for PATCH in BookMapper
